### ISSUE TITLE
Make Jaeger Client Cpp build and run on Windows

### ISSUE THREAD

**Issue by mdouaihy** [jaegertracing/jaeger-client-cpp#112]
Make the project available on windows. 

Currently, some header files, some cmake commands and a small part of the networking code make the build fail on Windows and MSVC 2017

**Comment by mdouaihy** [comment]
Hello,

I have a working build on https://github.com/mdouaihy/jaeger-client-cpp.

The main painpoints
* Difference of Network code between Windows/Linux
* Difference of API between Windows/Linux
* libraries link between Windows/Linux. Jaeger now links OpenTracing statically. On windows, this leads to multiple copy of opentracing in the final binary. This leads to UB especially when using the Global Tracer, which is a static variable in OT. To handle this, I changed the link to dynamic on Windows
* Different compilers. I used MSVC 15.7 which made way more easy the support of MSVC. Despite that, there is still some minor differences between the compilers
* Some tests are random by nature because they start a thread then sleep then expect some action to be done.

I need your feedback on it before I send PR.
